tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -28,110 +45,426 @@ test.describe( 'Merchant who is getting started', () => {
-		page.locator( "//button[text()='Connect'][not(@disabled)]" ).click();
-		await page.waitForLoadState( 'networkidle' );
-
-		// Expect the user to be redirected
-		await page.waitForURL( baseURL + 'auth_url' );
-	} );
-} );
+	test.describe( 'Connect WordPress.com account', () => {
+		test( 'should send an API request to connect Jetpack, and redirect to the returned URL', async ( {
+			baseURL,
+		} ) => {
+			// Mock Jetpack connect
+			await setUpAccountsPage.mockJetpackConnect( baseURL + 'auth_url' );
 
-test.describe( 'Merchant with Jetpack connected & Google not connected', () => {
-	test.beforeEach( async ( { page } ) => {
-		// Mock Jetpack as connected
-		await page.route( /\/wc\/gla\/jetpack\/connected\b/, ( route ) =>
-			route.fulfill( {
-				content: 'application/json',
-				headers: { 'Access-Control-Allow-Origin': '*' },
-				body: JSON.stringify( {
-					active: 'yes',
-					owner: 'yes',
-					displayName: 'testUser',
-					email: 'mail@example.com',
-				} ),
-			} )
-		);
-
-		// Mock google as not connected.
-		// When pending even WPORG will not render yet.
-		// If not mocked will fail and render nothing,
-		// as Jetpack is mocked only on the client-side.
-		await page.route( /\/wc\/gla\/google\/connected\b/, ( route ) =>
-			route.fulfill( {
-				content: 'application/json',
-				headers: { 'Access-Control-Allow-Origin': '*' },
-				body: JSON.stringify( {
-					active: 'no',
-					email: '',
-				} ),
-			} )
-		);
-
-		// Go to the setup page short way - directly via URL.
-		await page.goto(
-			'/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc'
-		);
-
-		// Wait for API calls and the page to render.
-		await page.waitForLoadState( 'networkidle' );
-		page.getByRole( 'heading', { name: 'Set up your accounts' } );
+			// Click the enabled connect button.
+			page.locator(
+				"//button[text()='Connect'][not(@disabled)]"
+			).click();
+			await page.waitForLoadState( 'networkidle' );
+
+			// Expect the user to be redirected
+			await page.waitForURL( baseURL + 'auth_url' );
+		} );
 	} );
 
-	test( 'should see their WPORG email, "Google" title & connect button', async ( {
-		page,
-	} ) => {
-		await expect( page.getByText( 'mail@example.com' ) ).toBeVisible();
+	test.describe( 'Connect Google account', () => {
+		test.beforeAll( async () => {
+			// Mock Jetpack as connected
+			await setUpAccountsPage.mockJetpackConnected(
+				'Test user',
+				'jetpack@example.com'
+			);
 
-		await expect(
-			page.getByText( 'Google', { exact: true } )
-		).toBeVisible();
+			// Mock google as not connected.
+			// When pending even WPORG will not render yet.
+			// If not mocked will fail and render nothing,
+			// as Jetpack is mocked only on the client-side.
+			await setUpAccountsPage.mockGoogleNotConnected();
 
-		expect(
-			page.getByRole( 'button', { name: 'Connect' } ).first()
-		).toBeEnabled();
+			await setUpAccountsPage.goto();
+
+			page.getByRole( 'heading', { name: 'Set up your accounts' } );
```

Comment:
I think this is useless here because there is no assertion possible. So even if the element is not found... it will continue with the tests.